notify/impl.go

```diff
@@ -62,6 +66,79 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// End-users should not be exposed to Go's type system,
+// as this will confuse them and prevent simple things like
+// simple equality checks to fail. Map everything to float64/string.
+type templateData struct {
+	Status                      string
+	Alerts                      []templateAlert
```

Comment:
> I would export both types for visibility in the godoc.

Done.

> I cannot immediately see rich use-cases that would cause much trouble here.

I can imagine users wanting to compare the labelname with the labelvalue, and also across with annotations.

The more common things that'll break will be string functions, they can only take in one type.